### PR TITLE
Missing autocommit for fs.open()

### PR DESCRIPTION
**Pull request by mariusvniekerk**
This adds in autocommit to the .open api.  Seems it was missing?

### CODE REVIEW

**Comment by martindurant** [comment]
Should be tested?

**Comment by martindurant** [comment]
Actually, implementing `discard` would be nice too (currently, if commit doesn't happen, the chunks will lie around until they eventually expire).

**Comment by mariusvniekerk** [comment]
Is discard part of the fsspec inferface?

**Comment by martindurant** [comment]
For files deriving from the AbstractBufferedFile
https://github.com/intake/filesystem_spec/blob/master/fsspec/spec.py#L887

File: s3fs/core.py [hunk 1]
```diff
@@ -959,13 +958,24 @@ def _initiate_upload(self):
...
+                    Bucket=self.bucket,
+                    Key=self.key, 
+                    PartNumber=1,
                     UploadId=self.mpu['UploadId'],
                     CopySource=self.path)
                 self.parts.append({'PartNumber': 1,
                                    'ETag': out['CopyPartResult']['ETag']})
 
+    @property
+    def bucket(self):
```
Comment (martindurant) [inline s3fs/core.py]: Just set these in `__init__`?

File: s3fs/core.py [hunk 2]
```diff
@@ -1073,6 +1081,16 @@ def commit(self):
                 self.fs.invalidate_cache(path)
             path = path + '/' + p
 
+    def discard(self):
+        if self.autocommit:
+            raise ValueError("Cannot discad when autocommit is enabled")
```
Comment (martindurant) [inline s3fs/core.py]: typo

**Comment by mariusvniekerk** [comment]
@martindurant updated

**Comment by martindurant** [comment]
Looks good, thanks!

**Pull request merged**